modules/module.base.ts

```diff
@@ -86,36 +86,44 @@ export class Module {
      * @param error The message of the error
      */
     handleError(error: string): any {
-        if(this.isHandleError)
+        if (this.isHandleError)
             throw new Error(error);
         return error;
     }
 
     /**
      * Simpilizing the response of the Module command
      * @param response The array response from the module
+     * @param returnSingleDimensionArray If we should return single dimension arrays without parsing them to objects
      */
-    handleResponse(response: any): any {
+    handleResponse(response: any, returnSingleDimensionArray = false): any {
```

Comment:
So instead of finding a way for it to work, you want to cancel the ability to parse it to objects?

Reply:
Before my modification weird objects were created, like:
```js
//From
[3, 'doc:3', 'doc:2', 'doc:1']
//To
{ 
    '3': 'doc:3',
    'doc:2': 'doc:1'
}
```
And I couldn't find any test case where the object mapping worked so I thought disabling it (only for search results) will be the best option.

I wanted to create mappings that return search queries like this:
```js
[
   'doc:1': {
        name: 'John Doe',
        age: 25,
        salary: 2500,
        introduction: 'John Doe is a developer at somekind of company.'
   },
   'doc:2': {
        name: 'Jane Doe',
        age: 30,
        salary: 5000,
        introduction: 'Jane Doe is John Doe\'s sister. She is not a developer, she is a hairstylist.'
   },
]
```

But that's outside of the scope of this PR. Unless you think that should be the correct fix for this @danitseitlin

Reply:
Well, I guess my attempt to create a dynamic conversion of the arrays into an object for easier use wasn't perfect. You can create specific mappings if you want, just for the relevant functions. @Legolaszstudio